controller/node_controller.go

```diff
@@ -323,9 +325,21 @@ func (nc *NodeController) syncDiskStatus(node *longhorn.Node) error {
 		}
 		// get disk available size
 		diskInfo, err := nc.getDiskInfoHandler(disk.Path)
-		if err != nil {
-			logrus.Errorf("Get disk information on node %v error: %v", node.Name, err)
+		// if the file system has been changed or the disk has been umount
+		if err != nil || diskInfo == nil || diskInfo.Fsid != diskID {
+			logrus.Errorf("File system on node %v has been changed: %v", node.Name, err)
+			// disable invalid disk
+			updateDisk.AllowScheduling = false
+			updateDisk.StorageMaximum = 0
+			updateDisk.StorageReserved = 0
+			diskStatus.State = types.DiskStateUnschedulable
+			diskStatus.StorageAvailable = 0
 		} else {
+			// if user has mount the disk back
```

Comment:
Don't need this comment. Just update the StorageMaximum. In fact, you can always update the StorageMaximum.

Also, don't update `AllowScheduling` to true automatically. I'd like to have the user do that manually since he/she manually mounts the disk back.